test/response-schema.spec.ts

```diff
@@ -219,4 +219,80 @@ describe('response schema', () => {
       expect(response.json()).toMatchInlineSnapshot();
     });
   });
+
+  describe('correctly process response schema and expects the non transformed type', () => {
+    let app: FastifyInstance;
+    beforeEach(async () => {
+      const REPLY_SCHEMA = z.object({
+        mySubObject: z
+          .object({
+            myDate: z.date(),
+          })
+          .transform(({ myDate }) => ({ myDate: myDate.toISOString() })),
+      });
+
+      app = Fastify();
+      app.setValidatorCompiler(validatorCompiler);
+      app.setSerializerCompiler(serializerCompiler);
+
+      app.after(() => {
+        app.withTypeProvider<ZodTypeProvider>().route({
+          method: 'GET',
+          url: '/',
+          schema: {
+            response: {
+              200: REPLY_SCHEMA,
+            },
+          },
+          handler: (req, res) => {
+            res.send({
+              mySubObject: {
+                // this is a Date as the input (aka. before transform) is a date
+                myDate: new Date('2021-05-12T00:00:00.000Z'),
+              },
+            });
+          },
+        });
+
+        app.withTypeProvider<ZodTypeProvider>().route({
+          method: 'GET',
+          url: '/incorrect',
+          schema: {
+            response: {
+              200: REPLY_SCHEMA,
+            },
+          },
+          handler: (req, res) => {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            res.send('test' as any);
+          },
+        });
+      });
+
+      await app.ready();
+    });
+    afterAll(async () => {
+      await app.close();
+    });
+
+    it('returns 200 for correct response', async () => {
```

Comment:
thank you for your contribution! can you explain the new test? which part of it would be failing before the change in this PR?

Reply:
I think the PR comment is answering this:

> On the provided test, we reshape a subkey from a Date to string with toISOString(). Without the PR, res.send expects myDate to be a string, which fails at runtime (the transform throws: String.prototype.toISOString() is not a function).
> With the change, res.send expects a Date, which passes at runtime :)

Is there anything that can be made clearer? The input schema has a transform, so the zod input type is a `{ myDate: Date }` and the zod output type `{ myDate: string }`
With this PR, you have to call `res.send(` with a `Date` and it correctly uses the transform to generate a string